argon2-jvm-nolibs/src/main/java/de/mkammerer/argon2/BaseArgon2.java

```diff
@@ -226,13 +229,50 @@ public HashResult hashAdvanced(int iterations, int memory, int parallelism, byte
 
         int result = Argon2Library.INSTANCE.argon2_hash(
                 jnaIterations, jnaMemory, jnaParallelism, password, new Size_t(password.length), salt, new Size_t(salt.length),
-                hash, new Size_t(hash.length), encoded, new Size_t(encoded.length), getType().getJnaType(), new JnaUint32(version.getJnaVersion())
+                hash, new Size_t(hash.length), encoded, new Size_t(encoded.length), getType().getJnaType(), version.getJnaType()
         );
         checkResult(result);
 
         return new HashResult(hash, Native.toString(encoded, ASCII));
     }
 
+    @Override
+    public byte[] rawHashAdvanced(int iterations, int memory, int parallelism, char[] password, Charset charset, byte[] salt, byte[] secret, byte[] associatedData) {
+        byte[] pwd = toByteArray(password, charset);
+        return rawHashAdvanced(iterations, memory, parallelism, pwd, salt, secret, associatedData, defaultHashLength, Argon2Version.NUMBER);
+    }
+
+    @Override
+    public byte[] rawHashAdvanced(int iterations, int memory, int parallelism, byte[] password, byte[] salt, byte[] secret, byte[] associatedData, int hashLength, Argon2Version version) {
+        int length = hashLength > 0 ? hashLength : defaultHashLength;
```

Comment:
Why this check? I would prefer to let argon2 raise an error instead of hiding it from the caller by using default hash lengths.

Reply:
As the caller has already the possibility to set a `defaultHashLength` while creating the `Argon2` object via the Factory I wanted to give the opportunity to also use this property when calculating the hash.

But I understand your opinion.

What would be your preferred method for this?
Simply skipping this check and let the C library throw an error via it's return code?
I would want to avoid to throw an exception which would require the caller to use a `try-catch`-block.

Reply:
By simply skipping the check this is what happens:
- Value `<=0`: JNA throws `IllegaglArgumentException` "Allocation size must be greater than zero"
- Value `0 < 16 <`: C library return code will bring `checkResult` to throw an `IllegalStateException` "Output is to short (-2)".

As these messages do not make clear which value was chosen incorrectly I would add this check instead:
```java
if (hashLength <= 0) throw new IllegalArgumentException("hashLength must be greater than zero");
```

Reply:
Yep, great, let's do your suggestion!

```
if (hashLength <= 0) throw new IllegalArgumentException("hashLength must be greater than zero");
```